go/testimonyd/internal/socket/socket.go

```diff
@@ -123,9 +123,14 @@ func (s *socket) String() string {
 // which the run() method passes to clients.
 func (s *socket) getNewBlocks() error {
 	blockIndex := 0
+	sleep := time.Millisecond
 	for {
 		b := s.blocks[blockIndex]
 		for !b.ready() {
+			time.Sleep(sleep)
```

Comment:
shouldn't sleep here; WaitForBlocks should sleep for up to 1sec as part of its 'select' call (see socket.cc)

Reply:
Yes, but I was testing (profiling) testimony with the stenographer (https://github.com/google/stenographer). And testimony creates high CPU utilization in this part of code when stenographer is working. I'll continue testing.